src/main/java/com/example/demo/matcher/Matcher.java

```diff
@@ -0,0 +1,86 @@
+package com.example.demo.matcher;
+
+import com.example.demo.matcher.models.Order;
+import com.example.demo.matcher.models.OrderAction;
+import com.example.demo.matcher.models.Trade;
+import com.example.demo.matcher.services.OrderService;
+import com.example.demo.matcher.services.TradeService;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class Matcher {
+    private final OrderService orderService;
```

Comment:
I might be wrong but could these be `@Autowired`? I would have a read about this here: https://www.baeldung.com/inversion-control-and-dependency-injection-in-spring . I think the best practice is constructor injection but I might be wrong. Let me know what you think about this :)